examples/keras_io/vision/image_classification_using_global_context_vision_transformer.py

```diff
@@ -0,0 +1,1188 @@
+        self.pool = keras.layers.GlobalAvgPool2D(name="pool")
+        self.head = keras.layers.Dense(num_classes, name="head", activation=head_act)
+
+    def build(self, input_shape):
+        super().build(input_shape)
+        self.built = True
+
+    def call(self, inputs, **kwargs):
+        x = self.patch_embed(inputs)  # shape: (B, H, W, C)
+        x = self.pos_drop(x)
+        for level in self.levels:
+            x = level(x)  # shape: (B, H_, W_, C_)
+        x = self.norm(x)
+        x = self.pool(x)  # shape: (B, C__)
+        x = self.head(x)
+        return x
+
+    def build_graph(self, input_shape=(224, 224, 3)):
+        """ref: https://www.kaggle.com/code/ipythonx/tf-hybrid-efficientnet-swin-transformer-gradcam"""
+        x = keras.Input(shape=input_shape)
+        return keras.Model(inputs=[x], outputs=self.call(x), name=self.name)
+
+    def summary(self, input_shape=(224, 224, 3)):
+        return self.build_graph(input_shape).summary()
+
+
+"""
+# Build Model
+* Let's build a complete model with all the modules that we've explained above. We'll
+build **GCViT-XXTiny** model with the configuration mentioned in the paper.
+* Also we'll load the ported official **pre-trained** weights and try for some
+predictions.
+"""
+
+# Model Configs
+config = {
+    "window_size": (7, 7, 14, 7),
+    "embed_dim": 64,
+    "depths": (2, 2, 6, 2),
+    "num_heads": (2, 4, 8, 16),
+    "mlp_ratio": 3.0,
+    "path_drop": 0.2,
+}
+ckpt_link = "https://github.com/awsaf49/gcvit-tf/releases/download/v1.1.5/gcvitxxtiny.weights.h5"
+
+# Build Model
+model = GCViT(**config)
+inp = ops.array(np.random.uniform(size=(1, 224, 224, 3)))
+out = model(inp)
+
+# Load Weights
+ckpt_path = keras.utils.get_file(ckpt_link.split("/")[-1], ckpt_link)
+model.load_weights(ckpt_path)
+
+# Summary
+model.summary((224, 224, 3))
+
+"""
+## Sanity check for Pre-Trained Weights
+"""
+
+img = keras.applications.imagenet_utils.preprocess_input(
+    chelsea(), mode="torch"
+)  # Chelsea the cat
+img = ops.image.resize(img, (224, 224))[None,]  # resize & create batch
+pred = model(img)
+pred_dec = keras.applications.imagenet_utils.decode_predictions(pred)[0]
+
+print("\n# Image:")
+plt.figure(figsize=(6, 6))
+plt.imshow(chelsea())
+plt.show()
+print()
+
+print("# Prediction (Top 5):")
+for i in range(5):
+    print("{:<12} : {:0.2f}".format(pred_dec[i][1], pred_dec[i][2]))
+
+"""
+# Fine-tune **GCViT** Model
+
+In the following cells, we will fine-tune **GCViT** model on Flower Dataset which
+consists `104` classes.
+"""
+
+"""
+## Configs
+"""
+
+# Model
+IMAGE_SIZE = (224, 224)
+
+# Hyper Params
+BATCH_SIZE = 32
+EPOCHS = 5
+
+# Dataset
+CLASSES = [
+    "dandelion",
+    "daisy",
+    "tulips",
+    "sunflowers",
+    "roses",
+]  # don't change the order
+
+# Other constants
+MEAN = 255 * np.array([0.485, 0.456, 0.406], dtype="float32")  # imagenet mean
+STD = 255 * np.array([0.229, 0.224, 0.225], dtype="float32")  # imagenet std
```

Comment:
Imagenet standardization isn't needed unless you're leveraging a pretrained model that expects it. Otherwise dividing pixel values by 255 is better

Reply:
GCVit indeed uses ImageNet mean. GCViT uses dataloader from `timm.data` which uses the ImageNet mean and std.
https://github.com/huggingface/pytorch-image-models/blob/b996c1a0f5068e7f5dfe69429e59e873536754c9/timm/data/constants.py#L3-L4